### DISCUSSION TITLE
Re-basing beets library?

### DISCUSSION THREAD

**Discussion by keeganjk** [beetbox/beets#3911] [Q&A]
I already have my library in a particular directory and some things should be saved in the beets library while others may not be (the file may have been deleted at some point to resolve some weird conflicts). What would be the best method to re-scan everything and move them to new directories (for example, I want to re-organize them by `$albumartist_sort` rather than `$albumartist`. Should I make a new directory for everything? I don't have space to have a temporary duplicate of everything. Would it be advisable to delete the saved library and just start over, maybe using `beet import -c` to move everything to its new directory?

Also, would the `-c` option move things as it goes, directory by directory, or would it move everything at once at the end of the process? Probably as it goes; that's how `beets` seems to work (this is preferred).

**Comment by sampsyo** [comment]
Hi! Does [this FAQ](https://beets.readthedocs.io/en/stable/faq.html#rename-my-files-according-to-a-new-path-format-configuration) help?

**Reply by keeganjk** [reply]
Sorry, it looks like it is on the right track, but I do not fully understand. Which syntax might I want to use to move all items in the database to their new location conforming to `config.yaml`? I do not exactly understand what is meant by `beet move [-capt] [-d DIR] QUERY`, e.g. is `-d DIR` supposed to be for the directory being moved or the directory being moved to, what is `QUERY`, etc.?

**Reply by keeganjk** [reply]
Never mind, I re-read and saw that it specified `-d` to be the destination directory. I am running the command now and it seems to be working properly.

**Reply by keeganjk** [reply]
After allowing it to run for a bit, it seems like it only moved a few (a small fraction of the total) of the items. Why might this be happening? I may have to move things manually and leave the ones with the old names.

**Reply by keeganjk** [reply]
In fact, after re-importing and trying to `move`, I was left with quite the mess, i.e. tracks and albums spread across multiple directories. I am cleaning it up now.
